cypress/integration/ResourcesForResearchers/index.js

```diff
@@ -0,0 +1,264 @@
+  } else if (title == "Tool Type") {
+    cy.get('.r4r-DEFAULT.home-nav__section').eq(0).should('contain.text', title);
+  } else if (title == "Research Area") {
+    cy.get('.r4r-DEFAULT.home-nav__section').eq(1).should('contain.text', title);
+  }
+});
+
+And('text {string} is displayed', (browseText) => {
+  const regex = new RegExp(browseText);
+  cy.get('.home-nav.r4r-DEFAULT p').invoke('text').should('match', regex);
+
+});
+
+And('{string} is a link', (viewAll) => {
+  const regex = new RegExp(viewAll);
+  cy.get('.r4r-DEFAULT.view-all__link').invoke('text').should('match', regex);
+
+});
+
+And('{string} section displays the following {string} categories', (section, options) => {
+
+  cy.document().then(document => {
+    let label;
+    if (section == "Tool Type") {
+      label = "Browse by Tool Type";
+    } else {
+      label = "Browse by Research Area";
+    };
+    const allAnchorTags = document.querySelectorAll(`[aria-label="${label}"]>div>a`);
+    for (let i = 0; i < allAnchorTags; i++) {
+      cy.get('[aria-label="${label}"]>div>a').eq(i).invoke('text').should('match', options[i]);
+    };
+  })
+
+});
+
+Then('the page options should appear at the bottom of the page', () => {
+  cy.get('.r4r-page-options-container').should('be.visible');
+});
+
+And('the following {string} are displayed', (pageOption) => {
+  const options = document.querySelectorAll(`div#PageOptionsControl1 > ul`);
+  for (let i = 0; i < options; i++) {
+    cy.get('div#PageOptionsControl1 > ul>li>a').should('contain.text', title);
+  }
+});
+
+And("user clicks on {string} link", (viewAll) => {
+  cy.get('.r4r-DEFAULT.view-all__link').click();
+});
+
+And("user is redirected to {string} with {string}", (redirectedPath, urlSearch) => {
+  cy.location('pathname').should('eq', redirectedPath);
+  cy.location('search').should('include', urlSearch);
+});
+
+And("{string} link has a href {string}", (linkText, linkHref) => {
+  cy.get('header > a').contains(linkText).should('have.attr', 'href').and('eq', linkHref);
+});
+
+And("search box is empty", () => {
+  cy.get('.searchbar__container input').should('contain.text', "");
+});
+
+And('results info text is {string}', (resultsInfo) => {
+  const regex = new RegExp(resultsInfo);
+  cy.get('[aria-label="Results count"] p').invoke('text').should('match', regex);
+});
+
+And('both pagers are displayed', () => {
+  cy.get('.r4r-results.r4r-DEFAULT>nav').should('have.length', 2);
+  cy.get('.r4r-results.r4r-DEFAULT>nav').each(($el) => {
+    expect('div r4r-pager__count  r4r-DEFAULT').to.exist;
+    expect('div r4r-pager__nav  r4r-DEFAULT').to.exist;
+  });
+
+});
+
+And('the results are displayed with each title containing a link', () => {
+  cy.get('[aria-label="search results"]').should('be.visible');
+  cy.get('[aria-label="search results"] > article>h2').first().should('be.visible').find('a').should('have.attr', 'href');
```

Comment:
the AC clearly says : ... **each** title containing a link - why are you only verifying the first one?

Reply:
duplicated locators again, please use alias

Reply:
```
 cy.get('section[aria-label="search results"]')
    .as("searchresults")
    .should("be.visible");
  cy.get("@searchresults")
    .find("article a")
    .each(($el) => {
      cy.wrap($el).invoke("text").should("not.be.empty");
      cy.wrap($el).should("have.attr", "href");
    });
```